pkg/prom/collector.go

```diff
@@ -1297,16 +1297,26 @@ func (c *SolidfireCollector) collectAsyncResults(ctx context.Context, ch chan<-
 		return err
 	}
 
+	mu.Lock()
+	defer mu.Unlock()
+
+	if len(ar.Result.AsyncHandles) == 0 {
+		ch <- prometheus.MustNewConstMetric(
+			MetricDescriptions.AsyncResultsActive,
+			prometheus.GaugeValue,
+			float64(0),
+			"no async results",
```

Comment:
I just realized this only helps for queries that aren't filtering by `ResultType` since those queries would still find no values due to this getting tagged with `no async results` instead of any ResultType.  But the problem is we don't know the full set of ResultTypes so we can't iterate over those and report a value of 0 for any result type not in the actual AsyncHandles result. Well that's not quite true, we can probably get a pretty good idea of the possible values for `ResultType` to hardcode but we might miss something if there's a result type that we've just never produced before.  I guess this might be the best we can/should do.

Reply:
Grabbed:  list from there https://docs.netapp.com/us-en/element-software/api/reference_element_api_listasyncresults.html#parameter